File: python/easy/find_the_highest_altitude.py
# https://leetcode.com/problems/find-the-highest-altitude/

class Solution:
    def largestAltitude(self, gain: List[int]) -> int:
        highest_alt = 0
        cur_alt = 0

        for alt in gain:
            cur_alt += alt
            highest_alt = max(highest_alt, cur_alt)

        return highest_alt

File: rust/easy/find_the_highest_altitude.rs
// https://leetcode.com/problems/find-the-highest-altitude/

use std::cmp::max;

impl Solution {
    pub fn largest_altitude(gain: Vec<i32>) -> i32 {
        let mut highest_alt = 0;
        let mut cur_alt = 0;

        for alt in gain.iter() {
            cur_alt += alt;
            highest_alt = max(highest_alt, cur_alt);
        }

        highest_alt
    }
}
